File: backend/app/schemas/team.py
"""T062: Team Pydantic DTOs with shift type integration."""

import uuid

from pydantic import BaseModel


class TimeWindowResponse(BaseModel):
    """Time window from shift type."""

    start: str
    end: str


class ShiftTypeDetail(BaseModel):
    """Shift type details for team response."""

    id: uuid.UUID
    name: str
    type: str
    time_windows: list[dict] | list[TimeWindowResponse]
    expected_hours: float
    total_hours: float
    uses_dynamic_close: bool
    description: str | None = None


class TeamCreate(BaseModel):
    """Create team with shift type reference."""

    name: str
    department: str
    shift_type_id: uuid.UUID  # FK to ShiftType


class TeamUpdate(BaseModel):
    """Update team fields (shift_type_id optional for changes)."""

    name: str | None = None
    department: str | None = None
    shift_type_id: uuid.UUID | None = None


class TeamMemberItem(BaseModel):
    id: uuid.UUID
    first_name: str
    last_name: str
    profile_image: str | None


class TeamResponse(BaseModel):
    """Team response with shift type details."""

    id: uuid.UUID
    name: str
    department: str
    shift_type_id: uuid.UUID
    # Shift type details
    shift_type: ShiftTypeDetail | None = None
    time_windows: list[dict] | None = None
    total_hours: float | None = None
    expected_hours: float | None = None
    uses_dynamic_close: bool | None = None
    # Team metadata
    is_active: bool
    members: list[TeamMemberItem] = []


class TeamMemberAdd(BaseModel):
    employee_id: uuid.UUID

File: frontend/src/types/team.ts
/**
 * T058: Team Types with Shift Type Integration
 */

import { ShiftType } from './shift-types';

export interface Team {
  id: string; // UUID
  name: string;
  department: string;
  shift_type_id: string; // FK to ShiftType
  shift_type?: ShiftType; // Full shift type details
  time_windows?: Array<{ start: string; end: string }>;
  total_hours?: number;
  expected_hours?: number;
  uses_dynamic_close?: boolean;
  is_active: boolean;
  members: TeamMember[];
  created_at: string;
  updated_at: string;
}

export interface TeamMember {
  id: string; // UUID
  first_name: string;
  last_name: string;
  profile_image?: string;
}

export interface TeamCreate {
  name: string;
  department: string;
  shift_type_id: string; // UUID reference to ShiftType
}

export interface TeamUpdate {
  name?: string;
  department?: string;
  shift_type_id?: string;
}

export interface TeamResponse {
  id: string;
  name: string;
  department: string;
  shift_type_id: string;
  shift_type?: ShiftType;
  time_windows?: Array<{ start: string; end: string }>;
  total_hours?: number;
  expected_hours?: number;
  uses_dynamic_close?: boolean;
  is_active: boolean;
  members: TeamMember[];
  created_at: string;
  updated_at: string;
}

export interface PaginatedTeams {
  items: TeamResponse[];
  total: number;
  page: number;
  pages: number;
  size: number;
}
